src/fixate/reporting/csv.py

```diff
@@ -115,45 +115,56 @@ class CSVWriter:
     def __init__(self):
         self.csv_queue = Queue()
         self.csv_writer = None
-        self.reporting = CsvReporting()
+
+        self.exception_in_test = False
+        self.failed = False
+        self.chk_cnt = 0
+        self.csv_path = ""
+        self.test_module = None
+        self.start_time = None
+        self.current_test = None
+        self.data = fixate.config.get_config_dict()
+        self.data.update(fixate.config.get_plugin_data("plg_csv"))
+        self.exception = None
 
     def install(self):
-        self.csv_writer = ExcThread(
-            target=self._csv_write, args=(self.csv_queue,), name="csv-writer"
-        )
+        self.csv_writer = ExcThread(target=self._csv_write, name="csv-writer")
         self.csv_writer.start()
 
+        pub.subscribe(self.test_start, "Test_Start")
+        pub.subscribe(self.test_comparison, "Check")
+        pub.subscribe(self.test_exception, "Test_Exception")
+        pub.subscribe(self.test_complete, "Test_Complete")
+        pub.subscribe(self.sequence_update, "Sequence_Update")
+        pub.subscribe(self.sequence_complete, "Sequence_Complete")
+        pub.subscribe(self.user_wait_start, "UI_block_start")
+        pub.subscribe(self.user_wait_end, "UI_block_end")
+        pub.subscribe(self.driver_open, "driver_open")
+
     def uninstall(self):
+        pub.unsubscribe(self.test_start, "Test_Start")
+        pub.unsubscribe(self.test_comparison, "Check")
+        pub.unsubscribe(self.test_exception, "Test_Exception")
+        pub.unsubscribe(self.test_complete, "Test_Complete")
+        pub.unsubscribe(self.sequence_update, "Sequence_Update")
+        pub.unsubscribe(self.sequence_complete, "Sequence_Complete")
+        pub.unsubscribe(self.user_wait_start, "UI_block_start")
+        pub.unsubscribe(self.user_wait_end, "UI_block_end")
+
```

Comment:
There is no unsubscribe for "driver_open" (which is a hangover from before)
how about putting the list of methods and topic string into a list and then iterating the list in install/unintsall?

Reply:
I'd also be happy to just fix the missing "driver_open" unsubscribed and not do the tidy up on this PR.